src/sentry/similarity/features.py

```diff
@@ -95,69 +93,96 @@ def extract(self, event):
                 )
         return results
 
-    def record(self, event):
+    def record(self, events):
+        if not events:
+            return []
+
+        scope = None
+        key = None
+
         items = []
-        for label, features in self.extract(event).items():
-            try:
-                features = map(self.encoder.dumps, features)
-            except Exception as error:
-                log = (
-                    logger.debug if isinstance(error, self.expected_encoding_errors) else
-                    functools.partial(logger.warning, exc_info=True)
-                )
-                log(
-                    'Could not encode features from %r for %r due to error: %r',
-                    event,
-                    label,
-                    error,
-                )
-            else:
-                if features:
-                    items.append((self.aliases[label], features, ))
+        for event in events:
+            for label, features in self.extract(event).items():
+                if scope is None:
```

Comment:
These assertions aren't the most elegant thing, but they work for now. Eventually it'd be nice to just collect these by `(scope, key)` and then route to the appropriate node(s) if there are more than one. This isn't a priority right now though